### PR TITLE
refactor: add server-side validation schema for events

### PR DESCRIPTION
**Pull request by elidrissidev**
Refactoring the events API endpoints to use server-side validation schema.

- See outline/outline#4284

### CODE REVIEW

File: server/routes/api/events/events.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      where = {
+        ...where,
+        [Op.or]: [
+          {
+            collectionId: collectionIds,
+          },
+          {
+            collectionId: {
+              [Op.is]: null,
+            },
```
Comment (elidrissidev) [inline server/routes/api/events/events.ts:73]: typescript started complaining here about `collectionId` not being nullable, hence why I made the change in Event model.

File: server/models/Event.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ class Event extends IdModel {
 
   @ForeignKey(() => Collection)
   @Column(DataType.UUID)
-  collectionId: string;
+  collectionId: string | null;
 
   @BelongsTo(() => Team, "teamId")
   team: Team;
```
Comment (elidrissidev) [inline server/models/Event.ts:94]: I wonder why these fields weren't nullable already, since I see from the database schema they are?

Comment (tommoor) [reply server/models/Event.ts:94]: 👍  Just a mistake

**Review by tommoor — APPROVED** [review]

File: server/models/Event.ts (hunk 2 above)
Comment (elidrissidev) [reply server/models/Event.ts:94]: alright, I will include that later.

**Comment by tommoor** [comment]
The database schema is probably a bit loose here, it's a very old model – there should always be an actor and a team at a minimum

**Comment by elidrissidev** [comment]
> The database schema is probably a bit loose here, it's a very old model – there should always be an actor and a team at a minimum

👍 just noticed the conditions inside the route handler. Should be good now.

**Review by tommoor — APPROVED** [review]

**Pull request merged**